crates/kitsune_p2p/kitsune_p2p/src/test_util/mock_network.rs

```diff
@@ -1,520 +0,0 @@
-            .send(KitsuneMock {
-                msg,
-                respond,
-                cert: connection.remote_cert.clone(),
-                url: connection.remote_url.clone(),
-            })
-            .await
-            .map_err(k_error)?;
-    } else if msg_id.is_res() {
-        if let Some(respond) = network.response_map.lock().remove(&msg_id.as_id()) {
-            respond.send(msg).map_err(k_error)?;
-        }
-    }
-    Ok(())
-}
-
-fn k_error<E: std::fmt::Debug>(e: E) -> KitsuneError {
-    format!("{:?}", e).into()
-}
-
-fn url_to_cert(url: &TxUrl) -> Tx2Cert {
-    Tx2Cert::from(
-        kitsune_p2p_types::tx_utils::ProxyUrl::from_full(url.as_str())
-            .expect("Mock network failed to parse url")
-            .digest(),
-    )
-}
-
-async fn read_response(connection: Arc<MockOutConnection>) -> KitsuneResult<(MsgId, PoolBuf)> {
-    let r = match connection.response_rx.recv().await {
-        Some(r) => match r.await {
-            Ok(r) => {
-                let KitsuneMockMsg { msg, id, mut buf } = r;
-
-                use kitsune_p2p_types::codec::Codec;
-                let data = msg.encode_vec().map_err(k_error)?;
-                buf.extend_from_slice(&data);
-                (id, buf)
-            }
-            Err(_) => {
-                return Err("end".into());
-            }
-        },
-        None => {
-            return Err("end".into());
-        }
-    };
-
-    Ok(r)
-}
-
-impl<T> SharedRecv<T> {
-    fn new(recv: mpsc::Receiver<T>) -> Self {
-        Self {
-            recv: Arc::new(parking_lot::Mutex::new(Some(recv))),
-        }
-    }
-
-    async fn recv(&self) -> Option<T> {
-        let mut recv = {
-            self.recv
-                .lock()
-                .take()
-                .expect("This type cannot be used concurrently")
-        };
-        let r = recv.recv().await;
-        *self.recv.lock() = Some(recv);
-        r
-    }
-
-    async fn select_recv<U>(&self, other: SharedRecv<U>) -> Option<Either<T, U>> {
-        let mut a = {
-            self.recv
-                .lock()
-                .take()
-                .expect("This type cannot be used concurrently")
-        };
-        let mut b = {
-            other
-                .recv
-                .lock()
-                .take()
-                .expect("This type cannot be used concurrently")
-        };
-        let r = {
-            let f_a = a.recv();
-            let f_b = b.recv();
-            futures::pin_mut!(f_a, f_b);
-            match futures::future::select(f_a, f_b).await {
-                Either::Left((t, _)) => t.map(Either::Left),
-                Either::Right((u, _)) => u.map(Either::Right),
-            }
-        };
-
-        *self.recv.lock() = Some(a);
-        *other.recv.lock() = Some(b);
-        r
-    }
-}
-
-impl<T> Clone for SharedRecv<T> {
-    fn clone(&self) -> Self {
-        Self {
-            recv: self.recv.clone(),
-        }
-    }
-}
```

Comment:
I think this also goes with the other mock_network module from Tom?

Reply:
That's tricky, I can't tell if they build on or with each other in any way... but even if they are entirely separate, the utility seems about the same.